content/projects/gsoc/2020/project-ideas/jenkins-distribution-customize-service.adoc

```diff
@@ -0,0 +1,41 @@
+---
+layout: gsocprojectidea
+title: "Jenkins distribution customize service"
+goal: "Provide an out of the box solution for Jenkins users"
+category: Service
+year: 2020
+status: draft
+skills:
+- Java
+- Web Development
+mentors:
+- "linuxsuren"
+---
+
+For many users, they download Jenkins first, then select some plugins and config them. 
+It might take a lot of time, like hours. But if we can get a perfect Jenkins distribution which contains all we need, 
+it can save that time for us.
+
+The service could be in the form of  a website offered as https://customize.jenkins.io. Plus, it should support self-host. 
+People can select the following configuration items:
+
+* Jenkins core version
+* plugins
+* common configuration, user/password, update-center site e.g.
+* plugin based configuration, Kubernetes, Sonarqube plugin config e.g.
+* multi-form package, jenkins.war or docker image
+* other things
```

Comment:
How about a backup/restore solution?

Reply:
Also, where to store Jenkins logs might interesting

Reply:
Agree. Other options also can be considered. https://jenkins.io/sigs/cloud-native/pluggable-storage/

Reply:
https://github.com/jenkinsci/aws-cloudwatch-logs-publisher-plugin